app/lib/presentation/pages/intro_page.dart

```diff
@@ -0,0 +1,170 @@
+
+  Future previousPage() async {
+    if (state.pageNumber == 0 && !isPop) {
+      isPop = true;
+      Navigator.of(context).maybePop();
+      return;
+    }
+    _introKey.currentState?.previous();
+  }
+
+  void onHorizontalDrag(DragEndDetails details) {
+    if (details.primaryVelocity == 0) {
+      return;
+    } else if (details.primaryVelocity!.compareTo(0) == -1) {
+      if (nextButton) {
+        nextPage();
+      }
+      return;
+    } else if (details.primaryVelocity!.compareTo(0) == 1) {
+      previousPage();
+    }
+  }
+
+  PageDecoration getPageDecoration() => const PageDecoration(
+        pageMargin: EdgeInsets.zero,
+        footerPadding: EdgeInsets.zero,
+        titlePadding: EdgeInsets.zero,
+        contentMargin: EdgeInsets.only(bottom: 30),
+      );
+
+  PageViewModel customPageViewModel({required Widget bodyWidget}) =>
+      PageViewModel(
+        useScrollView: false,
+        decoration: getPageDecoration(),
+        bodyWidget: bodyWidget,
+        titleWidget: const SizedBox(),
+      );
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData themeData = Theme.of(context);
+    final ColorScheme colorScheme = themeData.colorScheme;
+
+    return Scaffold(
+      body: PopScope(
+        canPop: state == IntroState.welcomePage,
+        onPopInvokedWithResult: (bool a, b) => previousPage(),
+        child: GestureDetector(
+          onHorizontalDragEnd: onHorizontalDrag,
+          child: IntroductionScreen(
+            isProgressTap: false,
+            key: _introKey,
+            dotsDecorator: DotsDecorator(
+              color: colorScheme.outlineVariant,
+              activeColor: colorScheme.primary,
+            ),
+            overrideNext: Center(
+              child: ButtonAtom(
+                variant: ButtonVariant.lowEmphasisIcon,
+                onPressed: nextButton ? nextPage : () {},
+                icon: FontAwesomeIcons.arrowRight,
+                disabled: !nextButton,
+              ),
+            ),
+            overrideBack: Center(
+              child: ButtonAtom(
+                variant: ButtonVariant.lowEmphasisIcon,
+                onPressed: previousPage,
+                icon: FontAwesomeIcons.arrowLeft,
+              ),
+            ),
+            overrideDone: Center(
+              child: ButtonAtom(
+                variant: ButtonVariant.highEmphasisFilled,
+                onPressed: onDone,
+                text: 'Skip',
+              ),
+            ),
+            pages: [
+              customPageViewModel(
+                bodyWidget: WelcomePage(),
+              ),
+              customPageViewModel(
+                bodyWidget: const PickLanguagePage(),
+              ),
+              customPageViewModel(
+                bodyWidget: const Center(
+                  child: Padding(
+                    padding: EdgeInsets.all(16.0),
+                    child: TextAtom('Choose a language page'),
+                  ),
```

Comment:
What about it?.
Don't give me half-baked pr.
Finish everything in the best way before I check